arangod/Agency/Agent.cpp

```diff
@@ -864,29 +871,32 @@ void Agent::advanceCommitIndex() {
   term_t t = _constituent.term();
   {
     WRITE_LOCKER(oLocker, _outputLock);
-    if (index > _commitIndex) {
+    if (index > _commitIndex.load(std::memory_order_relaxed)) {
 
       CONDITION_LOCKER(guard, _waitForCV);
       LOG_TOPIC("e24a9", TRACE, Logger::AGENCY)
-          << "Critical mass for commiting " << _commitIndex + 1 << " through "
+          << "Critical mass for commiting "
+          << _commitIndex.load(std::memory_order_relaxed) + 1 << " through "
           << index << " to read db";
       // Change _readDB and _commitIndex atomically together:
       _readDB.applyLogEntries(_state.slices(/* inform others by callbacks */
-                                            _commitIndex + 1, index),
-                              _commitIndex, t, true);
+                                            _commitIndex.load(std::memory_order_relaxed) + 1, index),
+                              _commitIndex.load(std::memory_order_relaxed), t, true);
```

Comment:
if `_commitIndex` has no external synchronization, then the two values of `_commitIndex` can be different here, and also different from the commit index value than was used for the comparison in the `if` above.
It would be better to load the `_commitIndex` value into a variable just once, and then use it for the if, the log message, and the function call to `applyLogEntries`.

Reply:
commitIndex is only changed under _outputLock. But still makes sense to load once.